.github/workflows/release.yml

```diff
@@ -0,0 +1,89 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - "v*"
+
+jobs:
+  create_release:
+    name: Create release
+    runs-on: ubuntu-latest
+    outputs:
+      upload_id: ${{ steps.draft_release.outputs.id }}
+      upload_url: ${{ steps.draft_release.outputs.upload_url }}
+    steps:
+      - name: Draft release
+        id: draft_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: Release ${{ github.ref }}
+          draft: true
+
+  build_release:
+    name: Build release
+    needs: create_release
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        include:
+          - os: ubuntu-latest
+            name: linux
+            dir_build: ./build
+            file_name: clap-info
+          - os: macos-latest
+            name: mac
+            dir_build: ./build
+            file_name: clap-info
+          - os: windows-latest
+            name: win
+            dir_build: ./build/Release
+            file_name: clap-info
+            file_ext: .exe
+    steps:
+      - name: Install Windows dependencies
+        if: matrix.os == 'windows-latest'
+        run: choco install zip
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Build binary
+        run: |
+          cmake -S . -B ./build
```

Comment:
Doesn’t this need a cmake build type of release at the first cmake for non multi config generators?

Reply:
Ah you're correct:
https://cmake.org/cmake/help/latest/guide/tutorial/Packaging%20Debug%20and%20Release.html

will add

Reply:
Great. Oh and on osx we will need the architecture flags also. (I would happily do this but am on vaca without a laptop right now).

-DCMAKE_OSX_ARCHITECTURES=“arm64;x86_64"

Reply:
If you ca. just add that I’m happy to merge from my phone and see what happens :)

and thanks again!

Reply:
done!

Reply:
Ok cool will merge!